### PR TITLE
feat(wash-cli): add --replace option to `wash app deploy`

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This commit adds an option `--replace` to `wash app deploy` which helps run repeated deploys of the *same* manifest by undeploying the specified model right before deploying.

While this isn't quite as robust as we want an option like `wash app dev` to be, it does enable easy scripting.

### Manual Verification

I tested this locally -- you can use the following from `crates/wash-cli`:

```
cargo run -- app deploy --replace <path to wadm.yaml>
```

### CODE REVIEW

**Comment by thomastaylor312** [comment]
Is this something that is handled by the new optional version stuff in wadm or is the purpose here more to force undeploy resources before deploying again?

**Comment by vados-cosmonic** [comment]
@thomastaylor312 this is more a precursor to [`wash app dev`](https://github.com/wasmCloud/wasmCloud/issues/1594) than to do with the version-less stuff, this enables re-deploying in one command without changing the manifest, which is a bit easier than removing the version and putting it back in later just before saving changes. This works *with* a version-less manifest and turns two commands to one still

File: crates/wash-cli/src/app/mod.rs [hunk 1]
```diff
@@ -216,6 +220,18 @@ async fn deploy_model(cmd: DeployCommand) -> Result<DeployModelResponse> {
         None => load_app_manifest("-".parse()?).await?,
     };
 
+    // If --replace was specified, we should attempt to replace the resources by deleting them beforehand
+    if cmd.replace {
+        if let (Ok(name), version) = (
+            get_model_name(&app_manifest),
+            get_model_version(&app_manifest).map(Into::into),
+        ) {
+            let _ =
```
Comment (thomastaylor312) [inline crates/wash-cli/src/app/mod.rs:229]: Nit: Might be worth logging a warning if this fails

File: crates/wash-cli/src/app/mod.rs [hunk 2]
```diff
@@ -268,6 +284,36 @@ async fn put_model(cmd: PutCommand) -> Result<PutModelResponse> {
...
+        }
+    }
+}
+
+/// Retrieve the version of a given [`AppManifest`], returning None if the manifest
+/// does not contain a version (or is not the type to contain a version)
+pub(crate) fn get_model_version(manifest: &AppManifest) -> Option<&str> {
+    match manifest {
+        AppManifest::ModelName(_) => None,
+        AppManifest::SerializedModel(manifest) => manifest
```
Comment (thomastaylor312) [inline crates/wash-cli/src/app/mod.rs:309]: Nit: I don't know if this returns the wadm `Manifest` type, but if it does, there is a [`version`](https://docs.rs/wadm-types/latest/wadm_types/struct.Manifest.html#method.version) method

**Review by thomastaylor312 — APPROVED** [review]

File: crates/wash-cli/src/app/mod.rs (hunk 2 above)
Comment (vados-cosmonic) [reply crates/wash-cli/src/app/mod.rs:309]: It does not! That's a plain YAML object (`serde_yaml::Value`) in there, unfortunately.

File: crates/wash-cli/src/app/mod.rs (hunk 1 above)
Comment (vados-cosmonic) [reply crates/wash-cli/src/app/mod.rs]: Ah this is a great suggestion, added!

**Pull request merged**